src/Illuminate/Database/Eloquent/Concerns/QueriesRelationships.php

```diff
@@ -202,6 +202,117 @@ public function withCount($relations)
     }
 
     /**
+     * Add subselect queries to sum the relations.
+     *
+     * @param  mixed  $relations
+     * @return $this
+     */
+    public function withSum($relations)
+    {
+        $relations = is_array($relations) ? $relations : func_get_args();
+
+        return $this->withAggregate($relations, 'SUM');
+    }
+
+    /**
+     * Add subselect queries to max the relations.
+     *
+     * @param  mixed  $relations
+     * @return $this
+     */
+    public function withMax($relations)
+    {
+        $relations = is_array($relations) ? $relations : func_get_args();
+
+        return $this->withAggregate($relations, 'MAX');
+    }
+
+    /**
+     * Add subselect queries to min the relations.
+     *
+     * @param  mixed  $relations
+     * @return $this
+     */
+    public function withMin($relations)
+    {
+        $relations = is_array($relations) ? $relations : func_get_args();
+
+        return $this->withAggregate($relations, 'MIN');
+    }
+
+    /**
+     * Add subselect queries to min the relations.
+     *
+     * @param  mixed  $relations
+     * @return $this
+     */
+    public function withAvg($relations)
+    {
+        $relations = is_array($relations) ? $relations : func_get_args();
+
+        return $this->withAggregate($relations, 'AVG');
+    }
+
+    /**
+     * use the MySQL aggregate functions including AVG COUNT, SUM, MAX and MIN.
+     *
+     * @param array $relations
+     * @param string $function
+     * @return $this
+     */
+    public function withAggregate($relations, $function = 'COUNT')
+    {
+        if (is_null($this->query->columns)) {
+            $this->query->select([$this->query->from.'.*']);
+        }
+
+        // set to lower
+        $function = Str::lower($function);
+
+        foreach ($this->parseWithRelations($relations) as $name => $constraints) {
+            // First we will determine if the name has been aliased using an "as" clause on the name
+            // and if it has we will extract the actual relationship name and the desired name of
+            // the resulting column. This allows multiple counts on the same relationship name.
+            $segments = explode(' ', $name);
+
+            unset($alias);
+
+            if (count($segments) == 3 && Str::lower($segments[1]) == 'as') {
+                list($name, $alias) = [$segments[0], $segments[2]];
+            }
+
+            // set the default column as * or primary key
+            $column = ($function == 'count') ? '*' : $this->model->getKeyName();
+
+            if (strpos($name, '|') !== false) {
```

Comment:
**Double negatives**
You can avoid the double negative of not, not having the pipe by using:

``` php
if (Str::contains($name, '|')) { /* ... */ }
```

**Pipe vs. comma**
And also I would personally choose a comma as a separator, since I find the pipe communicates `AND` whereas the comma communicates more separation than the pipe. Hope that makes sense.